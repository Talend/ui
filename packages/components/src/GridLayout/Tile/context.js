import React, { useContext } from 'react';
import invariant from 'invariant';

// eslint-disable-next-line import/prefer-default-export
export const TileContext = React.createContext();

export function useTileContext() {
	const context = useContext(TileContext);
	invariant(
		context,
		'@talend/react-components > Tile: you are using a sub component out of Tile.Container.',
	);
	return context;
}
