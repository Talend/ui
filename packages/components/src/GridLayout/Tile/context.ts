import { createContext, useContext } from 'react';

export type TileContextType = {
	displayMode: string;
	setDisplayMode: (mode: string) => void;
};

export const TileContext = createContext<TileContextType>({
	displayMode: '',
	// eslint-disable-next-line @typescript-eslint/no-unused-vars
	setDisplayMode: (mode: string) => {},
});

export function useTileContext() {
	const context = useContext(TileContext);
	return context;
}
