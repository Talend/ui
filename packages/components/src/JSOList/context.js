import React from 'react';

export const ListContext = React.createContext();
