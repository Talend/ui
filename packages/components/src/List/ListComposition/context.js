import React, { useContext } from 'react';

// eslint-disable-next-line import/prefer-default-export
export const ListContext = React.createContext();

export function useListContext() {
	const context = useContext(ListContext);
	if (!context) {
		throw Error(
			'@talend/react-components > List: you are using a sub component out of List.Manager.',
		);
	}
	return context;
}
