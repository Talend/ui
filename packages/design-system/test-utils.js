import React from 'react';
import { render } from '@testing-library/react';
import ThemeProvider from './src/components/ThemeProvider';

const AllTheProviders = ({ children }) => {
	return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
	const result = render(ui, { wrapper: AllTheProviders, ...options });
	result.getElement = () => result.container.childNodes[0];
	return result;
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
