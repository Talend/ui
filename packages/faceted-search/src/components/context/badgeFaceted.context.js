import { createContext, useContext } from 'react';
import invariant from 'invariant';

const badgeFacetedContext = createContext();

const useBadgeFacetedContext = () => {
	const context = useContext(badgeFacetedContext);
	invariant(
		context,
		'[badgeFacetedContext]: You are using a badge faceted component outside its context',
	);
	return context;
};

const BadgeFacetedProvider = badgeFacetedContext.Provider;

export { BadgeFacetedProvider, useBadgeFacetedContext };
