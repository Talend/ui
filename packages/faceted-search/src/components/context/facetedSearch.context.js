import { createContext, useContext } from 'react';
import invariant from 'invariant';

const facetedSearchContext = createContext();

const useFacetedSearchContext = () => {
	const context = useContext(facetedSearchContext);
	invariant(
		context,
		'[facetedSearchContext]: You are using one of faceted mode outside the faceted search',
	);
	return context;
};

const FacetedSearchProvider = facetedSearchContext.Provider;

export { FacetedSearchProvider, useFacetedSearchContext };
