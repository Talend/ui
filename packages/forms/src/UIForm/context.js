import React from 'react';

export const WidgetContext = React.createContext();

export function getWidget(widgets, widgetId, displayMode) {
	if (!widgets) {
		return undefined;
	}
	const id = displayMode ? `${widgetId}_${displayMode}` : widgetId;
	let widget = widgets[id];
	if (!widget) {
		widget = widgets[widgetId];
	}
	return widget;
}

export function useWidget(widgetId, displayMode) {
	const widgets = React.useContext(WidgetContext);
	return {
		widgets,
		WidgetImpl: getWidget(widgets, widgetId, displayMode),
	};
}
