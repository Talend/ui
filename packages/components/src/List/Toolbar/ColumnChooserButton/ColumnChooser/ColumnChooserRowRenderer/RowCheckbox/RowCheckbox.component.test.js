/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { ColumnChooserProvider } from '../../columnChooser.context';
import getDefaultT from '../../../../../../translate';

import Component from './RowCheckbox.component';

// eslint-disable-next-line react/prop-types
const RowCheckboxWithContext = ({ onChangeVisibility, id, t = getDefaultT(), ...rest }) => (
	<ColumnChooserProvider
		value={{
			onChangeVisibility,
			id,
			t,
		}}
	>
		<Component {...rest} />
	</ColumnChooserProvider>
);

describe('RowVisibilityCheckbox', () => {
	it('should render a checked checkbox input by default', () => {
		// Given
		const index = 0;
		// When
		const wrapper = mount(<RowCheckboxWithContext index={index} />);
		// Then
		expect(wrapper.find('input[type="checkbox"]')).toHaveLength(1);
		expect(wrapper.find('input[type="checkbox"]').prop('value')).toEqual(true);
		expect(wrapper.html()).toMatchSnapshot();
	});
	it('should render a non checked checked box input', () => {
		// Given
		const index = 0;
		const value = false;
		// When
		const wrapper = mount(<RowCheckboxWithContext index={index} value={value} />);
		// Then
		expect(wrapper.find('input[type="checkbox"]').prop('value')).toEqual(true);
	});
	it('should render a locked icon', () => {
		// Given
		const index = 0;
		const locked = true;
		// When
		const wrapper = mount(<RowCheckboxWithContext index={index} locked={locked} />);
		// Then
		expect(wrapper.find('svg').prop('name')).toEqual('talend-locked');
	});
	it('should call the onChangeVisibility when checkbox change and send the negate value', () => {
		// Given
		const index = 0;
		const onChangeVisibility = jest.fn();
		const value = false;
		// When
		const wrapper = mount(
			<RowCheckboxWithContext index={index} onChangeVisibility={onChangeVisibility} />,
		);
		act(() => {
			wrapper.find('input').simulate('change');
		});
		// Then
		expect(onChangeVisibility.mock.calls.length).toBe(1);
		expect(onChangeVisibility.mock.calls[0][0]).toBe(index);
		expect(onChangeVisibility.mock.calls[0][1]).toBe(!value);
	});
});
