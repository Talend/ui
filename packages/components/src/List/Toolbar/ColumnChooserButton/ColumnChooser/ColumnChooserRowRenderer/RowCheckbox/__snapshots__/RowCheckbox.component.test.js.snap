// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowCheckBox should render a checked checkbox input by default 1`] = `"<div id=\\"my-div-desc\\" class=\\"tc-column-chooser-aria-hidden theme-tc-column-chooser-aria-hidden\\">this is my checkbox</div>"`;
